### PR TITLE
Fixes export and execution of env variables for devfile commands

### LINKED ISSUE #4232: env variables in exec commands are not properly exported
/kind bug

## What versions of software are you using?

**Operating System:**

**Output of `odo version`:**

## How did you run odo exactly?

I have a devfile that defines a command spawning new shell, for example like this:

```yaml
  - id: build 
    exec:
      component: tools
      commandLine: "sh -c 'echo foo is $foo'"
      group:
        kind: build
        isDefault: true
      env:
        - name: foo
          value: bar
```

## Actual behavior

```
▶ odo push --show-log

Validation
 ✓  Validating the devfile [56469ns]

Creating Kubernetes resources for component java-maven
 ✓  Waiting for component to start [26s]

Applying URL changes
 ✓  URLs are synced with the cluster, no changes are required.

Syncing to component java-maven
 ✓  Checking file changes for pushing [1ms]
 ✓  Syncing files to the component [2s]

Executing devfile commands for component java-maven
 •  Executing build command "sh -c 'echo foo is $foo'"  ...
foo is
```

## Expected behavior

```▶ odo push --show-log

Validation
 ✓  Validating the devfile [56469ns]

Creating Kubernetes resources for component java-maven
 ✓  Waiting for component to start [26s]

Applying URL changes
 ✓  URLs are synced with the cluster, no changes are required.

Syncing to component java-maven
 ✓  Checking file changes for pushing [1ms]
 ✓  Syncing files to the component [2s]

Executing devfile commands for component java-maven
 •  Executing build command "sh -c 'echo foo is $foo'"  ...
foo is bar
```
notice the last line, it should print the correct value of $foo value

## more info
When command defines the env variable than odo executes it as `<env.name>=<env.value> && <commandLine>`.
But the correct behavior should be `export <env.name>=<env.value> && <commandLine>`. Export is important as it will make sure that the variable is available in all subshells not just in the current shell.

This is not just for `build` command, but for `run` and any other commands that have this behavior.

/priority high
/kind bug

### PR DESCRIPTION
**Pull request by mik-dass**
**What type of PR is this?**

/kind bug

**What does does this PR do / why we need it**:

Fixes export and execution of env variables for command in the devfile.

**Which issue(s) this PR fixes**:

Fixes #4232

**PR acceptance criteria**:

- [X] Unit test 

- [X] Integration test 

- [ ] Documentation 

- [ ] I have read the [test guidelines](https://github.com/openshift/odo/blob/master/docs/dev/test-architecture.adoc)

**How to test changes / Special notes to the reviewer**:

- Use a devfile that defines a command spawning a new shell, for example like this:
```
  - id: run 
    exec:
      component: tools
      commandLine: "sh -c 'mkdir $foo'"
      group:
        kind: run
        isDefault: true
      env:
        - name: foo
          value: bar
```
- Check if the command is executed successfully.

### CODE REVIEW

File: pkg/devfile/adapters/common/command_simple.go [hunk 1]
```diff
@@ -28,7 +28,10 @@ func newSimpleCommand(command devfilev1.Command, executor commandExecutor) (comm
 
 	// deal with environment variables
 	var setEnvVariable, cmdLine string
-	for _, envVar := range exe.Env {
+	for i, envVar := range exe.Env {
```
Comment (adisky) [inline pkg/devfile/adapters/common/command_simple.go:31]: should we make a small util function to do this?

Comment (dharmit) [reply pkg/devfile/adapters/common/command_simple.go:31]: +1, we're repeating this at multiple places. We should create a function for this.

**Review by dharmit — CHANGES_REQUESTED** [review]

**Comment by dharmit** [comment]
@mik-dass @kadel @adisky correct me if I'm doing something dumb here. 

Here's the devfile that I used:
```yaml
schemaVersion: 2.0.0
metadata:
  name: nodejs
  version: 1.0.0
  alpha.build-dockerfile: "https://raw.githubusercontent.com/odo-devfiles/registry/master/devfiles/nodejs/build/Dockerfile"
  alpha.deployment-manifest: "https://raw.githubusercontent.com/odo-devfiles/registry/master/devfiles/nodejs/deploy/deployment-manifest.yaml"
starterProjects:
  - name: nodejs-starter
    git:
      remotes:
        origin: "https://github.com/odo-devfiles/nodejs-ex.git"
components:
  - name: runtime
    container:
      image: registry.access.redhat.com/ubi8/nodejs-12:1-45
      memoryLimit: 1024Mi
      mountSources: true
      sourceMapping: /project
      endpoints:
        - name: http-3000
          targetPort: 3000
commands:
  - id: run 
    exec:
      component: runtime
      commandLine: "sh -c 'mkdir $foo'"
      group:
        kind: run
        isDefault: true
      env:
        - name: foo
          value: bar
```
And here's the `odo push` output:
```sh
$ odo push        

Validation
 ✓  Validating the devfile [34638ns]

Creating Kubernetes resources for component node-todo
 ✓  Waiting for component to start [6s]

Applying URL changes
 ✓  URL http-3000: http://http-3000-node-todo-buikmestmo.apps-crc.testing/ created

Syncing to component node-todo
 ✓  Checking files for pushing [680178ns]
 ✓  Syncing files to the component [243ms]

Executing devfile commands for component node-todo
 ✓  Executing run command "sh -c 'mkdir $foo'" [1s]

Pushing devfile component node-todo
 ✓  Changes successfully pushed to component
```
But if I spin up another shell, the env var doesn't seem to persist:
```sh
$ kubectl exec node-todo-5f5d5d5dd-7mcxm -- env | grep foo
ODO_COMMAND_RUN=export foo="bar" && sh -c 'mkdir $foo'

$ kubectl exec node-todo-5f5d5d5dd-7mcxm -- echo $foo
```
Am I doing or expecting something that's out of place or doing something wrong here?

File: pkg/devfile/adapters/common/command_simple.go (hunk 1 above)
Comment (mik-dass) [reply pkg/devfile/adapters/common/command_simple.go]: Fixed

**Comment by mik-dass** [comment]
> But if I spin up another shell, the env var doesn't seem to persist:

These env vars are only used during the execution of run, debug etc commands from the devfile.

**Review by dharmit — APPROVED** [review]
/approve
/lgtm

**Pull request merged**